dist/pom.xml

```diff
@@ -338,6 +338,23 @@
                             </target>
                         </configuration>
                     </execution>
+                    <execution>
+                        <id>check-shims-versions</id>
+                        <phase>package</phase>
```

Comment:
In the [scenario](https://github.com/NVIDIA/spark-rapids/pull/4265#issuecomment-985899707) from #4265, making it run `package` means that we hit the secondary problem first because bindary-dedupe is run inside build-parallel-worlds which is a `generate-resources` execution. On the other hand `check-shims-revisions.sh` relies on the build-parallel-worlds to unzip aggregator dependencies.

So I think to avoid a major refactoring and still get check-shims-revisions.sh called in time, we need to move its call to between the unzip loop and call to binary-dedupe https://github.com/NVIDIA/spark-rapids/blob/3e27acc6f91df5ab49e1ade8a1512985ec310b6d/dist/maven-antrun/build-parallel-worlds.xml#L120

Reply:
Good comment! 
Updated.